docs/4-language-usage/7-stored-objects/7-triggers/g-7730.md

```diff
@@ -0,0 +1,62 @@
+# G-7730: Avoid multiple DML events per trigger if primary key is assigned in trigger.
```

Comment:
Would the rule be still correct without "if primary key is assigned in trigger."? I have not tested it, but in the bad example we assign the PK only in the insert trigger. Can we assume that we assign a value to the PK in an insert trigger? I'd guess yes. 

I ask because I need to know that for the PL/SQL Cop validator. It has no idea what the primary key columns are. If this is a criterion then a check is not that simple to implement or might lead to a false positive.

Furthermore we should add an Oracle Database version for which we have tested this locking behaviour. One version is enough (latest) IMO.